docs/source/add_new_model.md

```diff
@@ -0,0 +1,573 @@
+This is the #1 motivation to re-implement models into 🤗Transformers - trying to maximize access 
+to a complex new NLP technology for **everybody**.
+	
+You should start thereby by diving into the original repository.
+
+#### Get familiar with the original repository.
+
+Successfully running the official pre-trained model in the original repository is often 
+**the most difficult** step. From our experience, it is very important to spend some time to get familiar with the original codebase. You should find out
+
+- Where to find the pre-trained weights
+- How to load the pre-trained weights into its corresponding model
+- Trace one forward pass so that you know which classes and functions are required for a simple forward pass
+  . Usually, you only have to reimplement those functions.
+- Be able to locate the important components of the model: Where is the model class? Are there submodel 
+  classes, *e.g.* EncoderModel, DecoderModel? Where is the self-attention layer? 
+  Are there multiple different attention layers, *e.g.* *self-attention*, *cross-attention*...?
+- How can you debug the model in the original environment of the repo? Do you have to set `print` statements
+  or can you work with an interactive debugger like `ipdb`?
+  
+It is very important that before you start opening a PR in 🤗Transformers that you can **efficiently** 
+debug code in the original repository! This means that you should be able to run a forward pass and print out the actual values of the output of a layer. *I.e* you can load a pre-trained model, pass an input vector of 
+token ids, *i.e.* `input_ids = [0, 1, 4, 5, ...]` to the model's forward function and you can print out the 
+intermediate outputs of - let's say - the first self-attention layer that could look something like this: 
+
+```bash
+[[
+ [-0.1465, -0.6501,  0.1993,  ...,  0.1451,  0.3430,  0.6024],
+ [-0.4417, -0.5920,  0.3450,  ..., -0.3062,  0.6182,  0.7132],
+ [-0.5009, -0.7122,  0.4548,  ..., -0.3662,  0.6091,  0.7648],
+ ...,
+ [-0.5613, -0.6332,  0.4324,  ..., -0.3792,  0.7372,  0.9288],
+ [-0.5416, -0.6345,  0.4180,  ..., -0.3564,  0.6992,  0.9191],
+ [-0.5334, -0.6403,  0.4271,  ..., -0.3339,  0.6533,  0.8694]]],
+```
+
+This means that your debugging environment should consists of a short script (ideally written by you) that 
+does the following (in pseudocode):
+
+```bash
+model = brand_new_bertModel.load_pretrained_checkpoint(/path/to/checkpoint/)
+input_ids = ... # vector of input ids
+original_output = model.predict(input_ids)
+```
+
+By running such a script, you should be able to print out intermediate values or hit a breakpoint
+in the clone of the original repository that is saved locally on your computer.
+
+We expect that every model added to 🤗Transformers passes a couple of integration tests, meaning that the original 
+model and the reimplemented version in 🤗Transformers have to give the exact same output up to a precision of 0.001! 
```

Comment:
I would explain why "0.001" too! Not necessarily the value you chose to showcase, but explaining that some difference can happen between frameworks and that's okay.